data/src/test/java/org/oppia/android/data/persistence/PersistentCacheStoreTest.kt

```diff
@@ -621,6 +622,7 @@ class PersistentCacheStoreTest {
   @Singleton
   @Component(
     modules = [
+      RobolectricProviderModule::class,
```

Comment:
Do we need this here?

Reply:
@anandwana001 TestDispatcherModule.kt contains provideTestCoroutineDispatchers and provideTestCoroutineDispatcherFactory that require `@IsOnRobolectric` annotation as a param 

so that might be the reason why we need to add RobolectricProviderModule for all Test files in which TestDispatcherModule is used

And there are very few test files that only require RobolectricProviderModule

Reply:
ah, ok. Is this separation causing any test fails?

Reply:
no